src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,497 @@
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+import * as opensearch from "aws-cdk-lib/aws-opensearchservice";
+import * as osis from "aws-cdk-lib/aws-osis";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as logs from "aws-cdk-lib/aws-logs";
+import { RemovalPolicy } from "aws-cdk-lib";
+import {  Stack } from "aws-cdk-lib";
+
+
+// Get the data stack
+const openSearchStack = Stack.of(backend.data);
+
+// Create the OpenSearch domain
+const openSearchDomain = new opensearch.Domain(
+  openSearchStack,
+  "OpenSearchDomain",
+  {
+    version: opensearch.EngineVersion.OPENSEARCH_2_11,
+    nodeToNodeEncryption: true,
+    encryptionAtRest: {
+      enabled: true,
+    },
+  }
+);
+
+```
+
+## Step 3: Setting Up Zero ETL from DynamoDB to OpenSearch
+
+### Step 3a: Setup Storage and IAM Role
+Establish Storage to back up raw events consumed by the OpenSearch pipeline. 
+Generate a file named `amplify/storage/resource.ts` and insert the provided content to set up a storage resource. Tailor your storage configurations to regulate access to different paths within your storage bucket.
+
+
+```ts title="amplify/storage/resource.ts"
+import { defineStorage } from "@aws-amplify/backend"
+
+export const storage = defineStorage({
+  name: "opensearch-backup-bucket-amplify-gen-2-test",
+  access: allow => ({
+    'public/*': [
+      allow.guest.to(['list', 'write', 'get'])
+    ]
+  })
+})
+
+```
+Retrieve the `s3BucketArn` and `s3BucketName` values from the storage resource as demonstrated below. Additionally, set up an IAM role for the pipeline and allocate the roles as specified. For additional details on the necessary IAM roles, please consult the [Setting up roles and users documentation](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/pipeline-security-overview.html#pipeline-security-create).
+
+```ts title="amplify/storage/resource.ts"
+import { defineStorage } from "@aws-amplify/backend"
+
+export const storage = defineStorage({
+  name: "opensearch-backup-bucket-amplify-gen-2-test",
+  access: allow => ({
+    'public/*': [
+      allow.guest.to(['list', 'write', 'get'])
+    ]
+  })
+})
+
+```
+Get the `s3BucketArn` and `s3BucketName` values from storage resource as shown below. Additionally, configure an IAM role for the pipeline and assign the roles as indicated below. For further information on the required IAM roles, please refer to the [Setting up roles and users](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/pipeline-security-overview.html#pipeline-security-create) documentation.
+
+``` ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
```

Comment:
to align with the other docs examples, the imports show similar as follows

```
import { Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
```

Reply:
this change requires code changes in multiple areas and testing take a significant amount of time. so not implementing this change